src/components/course/edit/sections/LecturerSection.vue

```diff
@@ -1,125 +1,121 @@
 <template>
-    <section class="border-t-2 py-8 border-gray-400">
-        <div class="lg:flex">
-            <div class="w-full lg:w-1/3 lg:block mr-12 flex flex-col mb-4">
-                <label class="block text-gray-700 text-md font-medium mb-2">Lecturer</label>
-                <label class="block text-gray-600"> Select a lecturer that will hold the course</label>
-            </div>
-            <div class="w-full lg:w-2/3">
-                <div class="mb-4 w-full">
-                    <label class="text-gray-700 text-md font-medium block mb-4">Lecturer-ID</label>
-                    <search-select
-                        v-model:selected="selectedOption"
-                        :options="optionsArray"
-                        input-id="lecturerId"
-                        :category-name="'Lecturer'"
-                    />
-                    <div :hidden="currentLecturer.username == ''" class="text-gray-700 text-md font-medium my-3">
-                        <label v-if="lecturerFound">
-                            <i class="text-green-400 fas fa-check mr-2"></i>
-                            {{ currentLecturer.firstName }} {{ currentLecturer.lastName }} (
-                            <router-link
-                                class="navigation-link cursor-pointer hover:underline"
-                                target="_blank"
-                                :to="{ name: 'profile.public', params: { username: currentLecturer.username } }"
-                            >
-                                @{{ currentLecturer.username }}
-                            </router-link>
-                            )
-                        </label>
-                        <label v-else>
-                            <i class="text-red-400 fas fa-times mr-2"></i>
-                            Lecturer-ID not found!
-                        </label>
-                    </div>
-                    <p v-if="errorBag.has('lecturerId')" class="error-message">
-                        {{ errorBag.get("lecturerId") }}
-                    </p>
+    <BaseSection subtitle="Select the lecturer that will hold the course." title="Lecturer">
+        <div class="lg:flex lg:space-x-12 lg:space-y-0 space-y-4 w-full">
+            <div class="lg:w-1/2 w-full">
+                <searchableSelect
+                    :id="'lecturerId'"
+                    label="Lecturer-ID"
+                    :placeholder="currentSelection.display"
+                    :elements="myLecturers"
+                    :selected="currentSelection"
+                    @update:selected="updateLecturer"
+                />
+                <label v-if="errorBag.has('lecturerId')" id="lecturerIdLabel" class="input-label-error">
```

Comment:
```suggestion
                <label v-if="errorBag.has('lecturerId')" id="lecturerIdErrorLabel" class="input-label-error">
```
I'd suggest to set the ID of error labels as such in general